packages/native-stack/src/views/useHeaderConfigProps.tsx

```diff
@@ -260,6 +306,74 @@ export function useHeaderConfigProps({
     </>
   );
 
+  const preparedHeaderLeftBarButtonItems = headerLeftItems
+    ? prepareHeaderBarButtonItems(headerLeftItems, route.key, 'left')
+    : undefined;
+  const preparedHeaderRightBarButtonItems = headerRightItems
+    ? prepareHeaderBarButtonItems(headerRightItems, route.key, 'right')
+    : undefined;
+  const hasHeaderBarButtonItems =
+    preparedHeaderLeftBarButtonItems?.length ||
+    preparedHeaderRightBarButtonItems?.length;
+
+  // Handle bar button item presses
+  const onPressHeaderBarButtonItem = hasHeaderBarButtonItems
+    ? (event: NativeSyntheticEvent<{ buttonId: string }>) => {
+        const pressedItem = [
+          ...(preparedHeaderLeftBarButtonItems ?? []),
+          ...(preparedHeaderRightBarButtonItems ?? []),
+        ].find(
+          (item) =>
+            item &&
+            'onPress' in item &&
+            item.buttonId === event.nativeEvent.buttonId
+        );
+        if (pressedItem && 'onPress' in pressedItem && pressedItem.onPress) {
+          pressedItem.onPress();
+        }
+      }
+    : undefined;
```

Comment:
Does this kind of code really need to be in React Navigation instead of React Native Screens? It doesn't sound ideal that `onPress` handling needs additional work on React Navigation side.